iroh-net/src/magicsock/endpoint.rs

```diff
@@ -712,18 +665,11 @@ impl Endpoint {
                 None => trace!(%peer, %ip_port, last_seen=%"never", "pruning address"),
             }
 
-            if let Some(addr_and_latency) = self.best_addr.as_ref() {
-                if addr_and_latency.addr == ip_port.into() {
-                    warn!(addr = %addr_and_latency.addr, "drop best_addr (prune for inactivity)");
-                    self.best_addr = None;
-                    // no longer relying on a direct connection, remove conn count
-                    inc!(MagicsockMetrics, num_direct_conns_removed);
-                    if self.derp_region.is_some() {
-                        // we now rely on a relay connection, add a relay count
-                        inc!(MagicsockMetrics, num_relay_conns_added);
-                    }
-                }
-            }
+            self.best_addr.clear_if_equals(
```

Comment:
this is a lot nicer but we lost the information of the peer whose best address is being dropped